leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```diff
@@ -0,0 +1,260 @@
+     */
+    private Arguments(List<Argument> argumentList) {
+        this.argumentList = argumentList;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Arguments [argumentList=%s]", argumentList.toString());
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((argumentList == null) ? 0 : argumentList.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Arguments other = (Arguments) obj;
+        if (argumentList == null) {
+            if (other.argumentList != null)
+                return false;
+        } else if (!argumentList.equals(other.argumentList))
+            return false;
+        return true;
+    }
+
+    /**
+     * Returns number of arguments.
+     */
+    public int size() {
+        return argumentList.size();
+    }
+
+    /**
+     * Check if Arguments has any elements.
+     */
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    /**
+     * Parse String into {@link Arguments}
+     *
+     * <pre>
+     * {@code
+     * arglist = arg *( "," arg )
+     * arg = DIGIT / DIGIT "=" "'" *CHAR "'"
+     * DIGIT = "0" / "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9"
+     * CHAR = "!" / %x23-26 / %x28-5B / %x5D-7E
+     * }
+     * </pre>
+     * 
+     * @param arglist String contains arguments to parse.
+     * @return {@link Arguments} represents list of arguments.
+     * @throws InvalidArgumentException if text has invalid format.
+     */
+    static Arguments parse(String arglist) throws InvalidArgumentException {
+        List<Argument> argumentList = new ArrayList<>();
+
+        if (arglist != null && !arglist.isEmpty()) {
+            int beginProbe = 0;
+            int validBegin = 0;
+            while (true) {
+                int separatorIndex = arglist.indexOf(',', beginProbe);
+                if (separatorIndex == -1) {
+                    break;
+                }
+                String argument = arglist.substring(validBegin, separatorIndex);
+
+                ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+                if (argumentParser.isValid()) {
+                    argumentList.add(argumentParser.parse());
+                    validBegin = separatorIndex + 1;
+                }
+                beginProbe = separatorIndex + 1;
+            }
+
+            String argument = arglist.substring(beginProbe);
+            ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+            argumentList.add(argumentParser.parse());
+        }
+
+        return new Arguments(argumentList);
+    }
+
+    private static class ArgumentParser {
```

Comment:
Maybe I'm wrong but the `ArgumentParser` class seems a bit over kill ?

Reply:
It's only for code logic separation. I'm not for very long classes with many methods. Usually I would put it into separated file but I don't want to generate too many files which would be harder to maintain.